spec/rubocop/cop/style/map_to_hash_spec.rb

```diff
@@ -46,11 +59,11 @@
         it 'registers an offense and corrects' do
           expect_offense(<<~RUBY, method: method)
             foo&.#{method} { |x| [x, x * 2] }.to_h
-                 ^{method} Pass a block to `to_h` instead of calling `#{method}&.to_h`.
+                 ^{method} Pass a block to `to_h` instead of calling `#{method}.to_h`.
           RUBY
 
           expect_correction(<<~RUBY)
-            foo.to_h { |x| [x, x * 2] }
+            foo&.to_h { |x| [x, x * 2] }
```

Comment:
This corrects typos in the original test case.

Reply:
@koic 

Now there is an error.

```ruby
2.7.6 :001 > foo = [1, 2, 3]
 => [1, 2, 3] 
2.7.6 :002 > foo&.map { |x| [x, x * 2] }.to_h
 => {1=>2, 2=>4, 3=>6} 
2.7.6 :003 > foo&.to_h { |x| [x, x * 2] }
 => {1=>2, 2=>4, 3=>6} 
```

```ruby
2.7.6 :005 > foo = nil
 => nil 
2.7.6 :006 > foo&.map { |x| [x, x * 2] }.to_h
 => {} 
2.7.6 :007 > foo&.to_h { |x| [x, x * 2] }
 => nil 
```